Source/Core/Core/PowerPC/Jit64Common/EmuCodeBlock.cpp

```diff
@@ -541,7 +542,8 @@ void EmuCodeBlock::SafeWriteRegToReg(OpArg reg_value, X64Reg reg_addr, int acces
 
   FixupBranch exit;
   const bool dr_set = (flags & SAFE_LOADSTORE_DR_ON) || m_jit.m_ppc_state.msr.DR;
-  const bool fast_check_address = !slowmem && dr_set && m_jit.jo.fastmem_arena;
+  const bool fast_check_address =
+      !slowmem && dr_set && m_jit.jo.fastmem_arena && !m_jit.m_ppc_state.m_enable_dcache;
```

Comment:
With the last commit, doesn't this also have to check `jo.fastmem`? Because fastmem itself could be turned off via the config but `jo.fastmem_arena` is now initialized always.

Reply:
This code only relies on the fastmem memory mappings, not on the exception handler. So it will work fine like this.

Reply:
JitArm64 might be broken, now that I think about it! I will mark this as a draft while I investigate.

Reply:
Aah okay I see, the `CheckIfSafeAddress()` checks for whether the load/write would generate an exception and the fastmem access is then only run if it won't throw, that makes sense. So basically, your last commit is changing the fastmem config option from affecting whether the fastmem region is mapped at all to only affecting whether we initialize the exception handler (unless the mapping fails I guess). I suppose that's fine, though it means that the no-mapping path (which I believe is used on iOS?) becomes more difficult to test. Might be worth adding an extra config option for that?

Reply:
> So basically, your last commit is changing the fastmem config option from affecting whether the fastmem region is mapped at all to only affecting whether we initialize the exception handler (unless the mapping fails I guess).

Actually, the exception handler gets initialized regardless. (We also need it for the BLR optimization.)

> I suppose that's fine, though it means that the no-mapping path (which I believe is used on iOS?) becomes more difficult to test. Might be worth adding an extra config option for that?

Yes, that doesn't sound like a bad idea.

Reply:
Actually, for now I'll just make the no-mapping path trigger whenever `fastmem` is false. The "check for whether the load/write would generate an exception" style of memory access would be possible to implement in JitArm64 for the `fastmem_arena && !fastmem` case, but it's non-trivial due to scratch register requirements, so I will leave it out of this PR.